plugins/net_plugin/net_plugin.cpp

```diff
@@ -1511,6 +1539,39 @@ namespace eosio {
       sync_wait();
    }
 
+   //-----------------------------------------------------------
+   void block_status_monitor::accepted() {
+      if( window_start_ != fc::time_point() ) {
+         window_start_ = fc::time_point();
+         events_ = 0;
+      }
+   }
+
+   void block_status_monitor::rejected() {
+      const auto now = fc::time_point::now();
+      // if starting a new window
+      if( window_start_ == fc::time_point() ) {
+         window_start_ = now;
+         return;
+      }
+
+      // if window time has not elapsed
+      const auto elapsed = now - window_start_;
+      if( elapsed < window_size_ ) {
+         return;
+      }
+
+      // if window time exactly expired
+      ++events_;
+      if( elapsed == window_size_ ) {
```

Comment:
This is extremely unlikely. Not sure I'm following the logic here.
accepted() should restart this monitor always.

Reply:
My thinking is that for consecutive windows, the end of one window is not necessarily the start of the next window.  To illustrate:
t0, the start of the first time window
te0, the time the first window closes
t1, the start of the next window
s, the window size
t, the time of the rejected block that closes the first window

te0 = t0 + s
t != te0
t1 = t instead of te0

The window_start is set to "0" in accepted().  However, the rejected() logic uses a window_start=0 as a signal to set the start of the next window.  My thinking is that this ensures each time window has at least 1 rejected block. My concern is that if the interval between rejected blocks is > than a multiple of the window size (s), a rejected block could result in the count being increased by more than one. This may be overthinking the problem.

Reply:
I think this overcomplicates it. Seems to me it should just be if you are in same window then don't increment; otherwise increment.